genesyscloud/flow_milestone/resource_genesyscloud_flow_milestone.go

```diff
@@ -0,0 +1,146 @@
+	sdkConfig := meta.(*gcloud.ProviderMeta).ClientConfig
+	proxy := getFlowMilestoneProxy(sdkConfig)
+
+	log.Printf("Reading flow milestone %s", d.Id())
+
+	return gcloud.WithRetriesForRead(ctx, d, func() *retry.RetryError {
+		flowMilestone, respCode, getErr := proxy.getFlowMilestoneById(ctx, d.Id())
+		if getErr != nil {
+			if gcloud.IsStatus404ByInt(respCode) {
+				return retry.RetryableError(fmt.Errorf("Failed to read flow milestone %s: %s", d.Id(), getErr))
+			}
+			return retry.NonRetryableError(fmt.Errorf("Failed to read flow milestone %s: %s", d.Id(), getErr))
+		}
+
+		cc := consistency_checker.NewConsistencyCheck(ctx, d, meta, ResourceFlowMilestone())
+
+		resourcedata.SetNillableValue(d, "name", flowMilestone.Name)
+		resourcedata.SetNillableReferenceWritableDivision(d, "division_id", flowMilestone.Division)
+		resourcedata.SetNillableValue(d, "description", flowMilestone.Description)
+
+		log.Printf("Read flow milestone %s %s", d.Id(), *flowMilestone.Name)
+		return cc.CheckState()
+	})
+}
+
+// updateFlowMilestone is used by the flow_milestone resource to update an flow milestone in Genesys Cloud
+func updateFlowMilestone(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*gcloud.ProviderMeta).ClientConfig
+	proxy := getFlowMilestoneProxy(sdkConfig)
+
+	flowMilestone := getFlowMilestoneFromResourceData(d)
+
+	log.Printf("Updating flow milestone %s", *flowMilestone.Name)
+	flowMilestoneSdk, err := proxy.updateFlowMilestone(ctx, d.Id(), &flowMilestone)
+	if err != nil {
+		return diag.Errorf("Failed to update flow milestone: %s", err)
+	}
+
+	log.Printf("Updated flow milestone %s", *flowMilestoneSdk.Id)
+	return readFlowMilestone(ctx, d, meta)
+}
+
+// deleteFlowMilestone is used by the flow_milestone resource to delete a flow milestone from Genesys cloud
+func deleteFlowMilestone(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*gcloud.ProviderMeta).ClientConfig
+	proxy := getFlowMilestoneProxy(sdkConfig)
+
+	_, err := proxy.deleteFlowMilestone(ctx, d.Id())
+	if err != nil {
+		return diag.Errorf("Failed to delete flow milestone %s: %s", d.Id(), err)
+	}
+
+	return gcloud.WithRetries(ctx, 180*time.Second, func() *retry.RetryError {
+		_, respCode, err := proxy.getFlowMilestoneById(ctx, d.Id())
+
+		if err != nil {
+			if gcloud.IsStatus404ByInt(respCode) {
+				log.Printf("Deleted flow milestone %s", d.Id())
+				return nil
+			}
+			return retry.NonRetryableError(fmt.Errorf("Error deleting flow milestone %s: %s", d.Id(), err))
+		}
+
+		return retry.RetryableError(fmt.Errorf("flow milestone %s still exists", d.Id()))
+	})
+}
+
+// getFlowMilestoneFromResourceData maps data from schema ResourceData object to a platformclientv2.Flowmilestone
+func getFlowMilestoneFromResourceData(d *schema.ResourceData) platformclientv2.Flowmilestone {
```

Comment:
I think this belongs in a utils file within this package. Probably seems like a lot to make a whole new file for one function, but better to stick with the same structure across the board